### DISCUSSION TITLE
Looking for guide to self-host/deploy Dify on local server

### DISCUSSION THREAD

**Discussion by roycordero** [langgenius/dify#8872] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

Hi everyone,

I am looking to deploy Dify on my local server correctely for use in my bussines as producction service. 

my server is a proxmox (services and vm virtual machines) orchestrator and my dify is running ina vm with ubuntu 22.04 with the minimun especs: 2cores, 4gb ram 

Could anyone provide me an advise or a detailed guide or best practices for correctly deploying Dify on local server for production? 

any help could be a great thanks

### 2. Additional context or comments

* im not using any gpu at the moment
* i want to integrate dify with another apps using the API
* i want to use a model like Llama or qwen self hosted as well

**Comment by y-omr** [comment]
Hi 🖐️
I'm not a native English speaker, so if I misunderstand your question and my answer isn't what you're looking for, I apologize.
We believe that deploying with Docker Compose is the simplest way to do this.

https://docs.dify.ai/getting-started/install-self-hosted/docker-compose

>i want to integrate dify with another apps using the API

Here’s how you can call workflows created in Dify via the API:
https://docs.dify.ai/guides/application-publishing/developing-with-apis

> i want to use a model like Llama or qwen self hosted as well

How about hosting the LLM model on Ollama and integrating Dify with Ollama?
https://docs.dify.ai/development/models-integration/ollama
